pygit2/repository.py

```diff
@@ -709,10 +717,16 @@ def merge_file_from_index(
         )
         check_error(err)
 
-        ret = ffi.string(cmergeresult.ptr, cmergeresult.len).decode('utf-8')
+        automergeable = cmergeresult.automergeable != 0
+        content = ffi.string(cmergeresult.ptr, cmergeresult.len).decode('utf-8')
+        filemode = cmergeresult.mode
+        path = cmergeresult.path
         C.git_merge_file_result_free(cmergeresult)
 
-        return ret
+        if not return_full:
+            return content
+
+        return automergeable, content, filemode, path
```

Comment:
Could you return a smarter object, like a dataclass? With the name `MergeFileResult`, and using "contents" instead of "content" (as is written in the libgit2 reference documentation).

Reply:
Yep... will do both things. Thanks for the feedback.